sdk_extensions/zpages/README.md

```diff
@@ -2,8 +2,100 @@
 
 [![Javadocs][javadoc-image]][javadoc-url]
 
-This module contains code for OpenTelemetry's Java zPages.
+This module contains code for OpenTelemetry's Java zPages, which are a collection of dynamic HTML
+web pages that display stats and trace data.
+
+* Java 7 compatible.
 
 <!--- TODO: Update javadoc -->
 [javadoc-image]: https://www.javadoc.io/badge/io.opentelemetry/opentelemetry-sdk-contrib-auto-config.svg
 [javadoc-url]: https://www.javadoc.io/doc/io.opentelemetry/opentelemetry-sdk-contrib-auto-config
+
+## Quickstart
+
+### Add the dependencies to your project
+
+For Maven, add the following to your `pom.xml`:
+```xml
+<dependencies>
+  <dependency>
+        <groupId>io.opentelemetry</groupId>
+        <artifactId>opentelemetry-api</artifactId>
+        <version>0.7.0</version>
+      </dependency>
+      <dependency>
+        <groupId>io.opentelemetry</groupId>
+        <artifactId>opentelemetry-sdk</artifactId>
+        <version>0.7.0</version>
+      </dependency>
+      <dependency>
+        <groupId>io.opentelemetry</groupId>
+        <artifactId>opentelemetry-sdk-extension-zpages</artifactId>
+        <version>0.7.0</version>
+      </dependency>
+</dependencies>
+```
+
+<!--- TODO: Verify gradle configuration -->
+For Gradle, add the following to your dependencies:
+```groovy
+api 'io.opentelemetry:opentelemetry-api:0.7.0'
+api 'io.opentelemetry:opentelemetry-sdk:0.7.0'
+implementation 'io.opentelemetry:opentelemetry-sdk-extension-zpages:0.7.0'
+```
+
+### Register the zPages
+
+**Note:** The package `com.sun.net.httpserver` is required to use zPages. Please make sure your
```

Comment:
```suggestion
**Note:** The package `com.sun.net.httpserver` is required to use the default zPages setup. Please make sure your
```

It's technically not required if users mount the pages to their own server, which hopefully will be fairly common.

Reply:
`private static HttpServer server;` (`import com.sun.net.httpserver.HttpServer`). 
Even with a variable like this, if the users mount to their own server, `com.sun.net.httpserver` is not required anymore?

Reply:
Java is surprisingly lazy, but indeed because it's static I think the answer is no. Can we go ahead and fix it so it's not required? It just means separating out a class that manages the span processor and having only a small class dealing with the server

Reply:
Sorry I didn't entire get what you meant, should we separate the current `ZPageServer` class somehow?

Reply:
Yeah I think if we separate initialization of span processors to a separate class, the hard dependency on sun server can be avoided. But that can be a follow-up for after this PR.

Reply:
Quick question regarding making HttpServer not hard required. Should I separate out
```
private static HttpServer server;
...
public static void startHttpServerAndRegisterAllPages(int port) throws IOException {}
```
to a different file. And keep the register function
```
public static void registerAllPagesToHttpServer(HttpServer server) {}
```

Will the HttpServer type parameter still make sun package hard required?

Reply:
I think the method users will need when not using `HttpServer` is

https://github.com/open-telemetry/opentelemetry-java/blob/master/sdk_extensions/zpages/src/main/java/io/opentelemetry/sdk/extensions/zpages/ZPageServer.java#L91

So if that was moved to a separate class the hard dependency should be gone I believe.

Reply:
How do they use zPages at all then with the current implementation?

Reply:
Haven't actually compiled this code but for example, if using Armeria, it would look something like this

```java

class ZPageService implements HttpService {
  private final ZPageHandler handler;

  public HttpResponse service(ServiceRequestContext ctx, HttpRequest req) {
    Map<String, String> params = ctx.routingContext().queryParams().stream().collect(toMap());
    ByteBufOutputStream payload = new ByteBufOutputStream();
    handler.emitHtml(params, payload);
    return HttpResponse.of(RequestHeaders.of(HttpStatus.OK), HttpData.wrap(payload.buf()));
  }
}
ZPages.registerSpanProcessors();

Server server = Server.builder();
for (ZPageHandler handler : ZPages.getAllHandlers()) {
  server.service(handler.getUrlPath(), new ZPageService(handler);
}
```

Very efficient zpages, all on the same port as my service :)